packages/api/cms-api/src/content-generation/generate-alt-text.resolver.ts

```
@@ -0,0 +1,17 @@
+import { Inject } from "@nestjs/common";
+import { Args, Mutation, Resolver } from "@nestjs/graphql";
+
+import { RequiredPermission } from "../user-permissions/decorators/required-permission.decorator";
+import { CONTENT_GENERATION_SERVICE } from "./content-generation.constants";
+import { ContentGenerationServiceInterface } from "./content-generation.service.interface";
+
+@Resolver()
+export class GenerateAltTextResolver {
+    @Inject(CONTENT_GENERATION_SERVICE) private contentGenerationService: ContentGenerationServiceInterface;
+
+    @RequiredPermission(["dam"], { skipScopeCheck: true })
+    @Mutation(() => String)
+    async generateAltText(@Args("fileUrl", { type: () => String }) fileUrl: string): Promise<string> {
+        return this.contentGenerationService.generateAltText?.(fileUrl) ?? "";
```

We should throw an error here if the method isn't implemented